### PR TITLE
Fix typos

### PR DESCRIPTION
**Pull request by cassella**
Fix typos in scopeResolve.cpp, functionResolution.cpp, ChapelBase.chpl,
String.chpl, Lists.chpl, DecoratedClasses.h, scopeResolve.cpp,
and one in SharedObject.chpl and a .good file.

### CODE REVIEW

File: modules/internal/ChapelBase.chpl [hunk 1]
```diff
@@ -840,7 +840,7 @@ module ChapelBase {
     }
   }
 
-  pragma "unsafe" // work around problems storting non-nilable classes
+  pragma "unsafe" // work around problems storing non-nilable classes
```
Comment (cassella) [inline modules/internal/ChapelBase.chpl]: I'm 90% sure this should be "storing", not "sorting".  (Or "starting" or "snorting".)

**Comment by cassella** [comment]
@mppf Would you care to weigh in on "nilability" vs "nilibility"?

```
fortytwo@magrathea:~/src/chapel (typos)$ grep -ri nilability compiler modules | wc -l
31
fortytwo@magrathea:~/src/chapel (typos)$ grep -ri nilibility compiler modules | wc -l
8
```

The internet seems to favor the former.

**Comment by mppf** [comment]
nilability sounds good to me. thanks for helping with thes.

**Comment by cassella** [comment]
Rebased to master, fixed more typos, and reran the test whose .good file changed.

**Pull request merged**